### PR TITLE
Render indirect calls directly in no-sandbox mode.

### PR DESCRIPTION
**Pull request by matthias-blume**
This removes the use of the CALL_INDIRECT macro in no-sandbox mode.
It also avoids mentioning any tables or any function type variables.  These will go away in a separate PR.

### CODE REVIEW

File: src/c-writer.cc [hunk 1]
```diff
@@ -2710,20 +2710,32 @@ void CWriter::Write(const ExprList& exprs) {
...
+
+          Write("CALL_INDIRECT(",
+                ExternalInstanceRef(ModuleFieldType::Table, table->name), ", ");
+          WriteCallIndirectFuncDeclaration(decl, "(*)");
+          Write(", ", FuncTypeExpr(func_type), ", ", StackVar(0));
+          Write(", ", ExternalInstanceRef(ModuleFieldType::Table, table->name),
+                ".data[", StackVar(0), "].module_instance");
+          if (num_params > 0) {
+            Write(", ");
+          }
```
Comment (dimitry-) [inline src/c-writer.cc]: personal preference: maybe do `needs_comma = true`, and check it inside the loop below? As it is it assumes knowledge of now the code below is written which is generally is undesirable.

**Review by dimitry- — APPROVED** [review]

File: src/c-writer.cc (hunk 1 above)
Comment (matthias-blume) [reply src/c-writer.cc]: done

File: src/c-writer.cc [hunk 2]
```diff
@@ -2727,12 +2728,10 @@ void CWriter::Write(const ExprList& exprs) {
...
-          if (num_params > 0) {
-            Write(", ");
-          }
+          needs_initial_comma = true;
         }
         for (Index i = 0; i < num_params; ++i) {
-          if (i > 0) {
+          if (needs_initial_comma || i > 0) {
             Write(", ");
           }
```
Comment (dimitry-) [inline src/c-writer.cc:2738]: you can also do `else { needs_comma = true; }` and drop `i > 0` above.

Comment (matthias-blume) [reply src/c-writer.cc:2738]: done

**Pull request merged**